agent/lib_aws_sdk_php.c

```diff
@@ -14,10 +14,409 @@
+
+/*
+* Note: For SQS, the command_arg_array will contain the following arrays seen
+below:
+//clang-format off
+$result = $client->receiveMessage(array(
+    // QueueUrl is required
+    'QueueUrl' => 'string',
+    'AttributeNames' => array('string', ... ),
+    'MessageAttributeNames' => array('string', ... ),
+    'MaxNumberOfMessages' => integer,
+    'VisibilityTimeout' => integer,
+    'WaitTimeSeconds' => integer,
+));
+
+$result = $client->sendMessage(array(
+    // QueueUrl is required
+    'QueueUrl' => 'string',
+    // MessageBody is required
+    'MessageBody' => 'string',
+    'DelaySeconds' => integer,
+    'MessageAttributes' => array(
+        // Associative array of custom 'String' key names
+        'String' => array(
+            'StringValue' => 'string',
+            'BinaryValue' => 'string',
+            'StringListValues' => array('string', ... ),
+            'BinaryListValues' => array('string', ... ),
+            // DataType is required
+            'DataType' => 'string',
+        ),
+        // ... repeated
+    ),
+));
+
+$result = $client->sendMessageBatch(array(
+    // QueueUrl is required
+    'QueueUrl' => 'string',
+    // Entries is required
+    'Entries' => array(
+        array(
+            // Id is required
+            'Id' => 'string',
+            // MessageBody is required
+            'MessageBody' => 'string',
+            'DelaySeconds' => integer,
+            'MessageAttributes' => array(
+                // Associative array of custom 'String' key names
+                'String' => array(
+                    'StringValue' => 'string',
+                    'BinaryValue' => 'string',
+                    'StringListValues' => array('string', ... ),
+                    'BinaryListValues' => array('string', ... ),
+                    // DataType is required
+                    'DataType' => 'string',
+                ),
+                // ... repeated
+            ),
+        ),
+        // ... repeated
+    ),
+));
+
+//clang-format on
+*/
+void nr_lib_aws_sdk_php_sqs_handle(nr_segment_t* segment,
+                                   char* command_name_string,
+                                   size_t command_name_len,
+                                   NR_EXECUTE_PROTO) {
+  char* command_arg_value = NULL;
+  nr_segment_message_params_t message_params = {
+      .library = "SQS",
+      .destination_type = NR_MESSAGE_DESTINATION_TYPE_QUEUE,
+      .messaging_system = "aws_sqs",
+  };
+  nr_segment_cloud_attrs_t cloud_attrs = {0};
+
+  if (NULL == segment) {
+    return;
+  }
+
+  if (NULL == command_name_string || 0 == command_name_len) {
+    return;
+  }
+
+  /* Determine if we instrument this command. */
+  if (command_name_len == AWS_SQS_SEND_MESSAGE_BATCH_COMMAND_LEN
+      && 0
+             == nr_strncmp(AWS_SQS_SEND_MESSAGE_BATCH_COMMAND,
+                           command_name_string,
+                           AWS_SQS_SEND_MESSAGE_BATCH_COMMAND_LEN)) {
```

Comment:
What is the advantage of using `nr_strncmp` over `nr_strncmp` or `nr_streq`?

Reply:
strncmp was needed since the defined strings had an extra char on the end that we could account for if we used the precomputed length instead of the length computed in nr_streq.
This could be made more readable with a [macro](https://github.com/newrelic/newrelic-php-agent/pull/1003#discussion_r1915814849).

Reply:
What is the extra char on the end in the defined string that needs to be accounted for?

Reply:
With the macro added, nrstrcmp is no longer needed.

defined strings and lengths were also removed.
See the changes:
[fa419e3](https://github.com/newrelic/newrelic-php-agent/pull/1003/commits/fa419e3526c46177e60f761426cfe74e5f799b80)
[7d9c4c0](https://github.com/newrelic/newrelic-php-agent/pull/1003/commits/7d9c4c03123dad8403b1268d85a8af004da51129)